onnx/__init__.py

```diff
@@ -98,7 +112,14 @@ def load_model(f, format=None):  # type: (Union[IO[bytes], Text], Optional[Any])
     Loaded in-memory ModelProto
     '''
     s = _load_bytes(f)
-    return load_model_from_string(s, format=format)
+    model = load_model_from_string(s, format=format)
+
+    model_filepath = _get_file_path(f)
+    if model_filepath:
+        basepath = os.path.dirname(model_filepath)
+        model = add_basepath_to_external_data_tensors(model, basepath)
+
+    return model
```

Comment:
for `load_model` we should have another parameter `load_external_data` (default should be true) to let user decide whether loading the external data or not. 

Also we should have another helper function `load_external_data_for_model(model, base_dir)`. I think we need to make it clear that it's users' responsibility to keep the base dir info. It's kind of ugly to save extra info and strip it out after loading.

Reply:
Agreed. We can also make it more flexible by adding more methods and parameters.

We should just make sure we keep the user interface simple and backwards compatible. A simple call to `onnx.load_model(path)` and `onnx.save_model(proto, path)` should work whether data is internally or externally stored.